snippets/base/tests/test_admin.py

```diff
@@ -3,47 +3,35 @@
 
 from mock import Mock, patch
 
-from snippets.base.admin import SnippetAdmin, SnippetTemplateAdmin
-from snippets.base.models import Snippet, SnippetTemplate, SnippetTemplateVariable
-from snippets.base.tests import SnippetTemplateFactory, SnippetTemplateVariableFactory, TestCase
+from snippets.base.admin import SnippetTemplateAdmin
+from snippets.base.models import SnippetTemplate, SnippetTemplateVariable
+from snippets.base.tests import (SnippetFactory, SnippetTemplateFactory,
+                                 SnippetTemplateVariableFactory, TestCase)
 
 
 class SnippetAdminTests(TestCase):
     def setUp(self):
         self.factory = RequestFactory()
-        self.model_admin = SnippetAdmin(Snippet, None)
-        self.model_admin.admin_site = Mock()
-        self.user = User.objects.get_or_create(username='foo', email='foo@example.com')[0]
-
-    def test_save_as_published(self):
-        request = self.factory.post('/', data={
-            'name': 'test',
-            'template': 'foo',
-            'published': u'on',
-            '_saveasnew': True
-        })
-        request.user = self.user
-
-        with patch('snippets.base.admin.admin.ModelAdmin.change_view') as change_view_mock:
-            self.model_admin.change_view(request, 999)
-            change_view_mock.assert_called_with(request, 999, '', None)
-            request = change_view_mock.call_args[0][0]
-            self.assertEqual(request.POST['published'], u'off')
-
-    def test_normal_save_published(self):
-        """Test that normal save doesn't alter 'published' attribute."""
-        request = self.factory.post('/', data={
-            'name': 'test',
-            'template': 'foo',
-            'published': u'foo'
-        })
-        request.user = self.user
-
-        with patch('snippets.base.admin.admin.ModelAdmin.change_view') as change_view_mock:
-            self.model_admin.change_view(request, 999)
-            change_view_mock.assert_called_with(request, 999, '', None)
-            request = change_view_mock.call_args[0][0]
-            self.assertEqual(request.POST['published'], u'foo')
+        User.objects.create_superuser(username='admin',
+                                      email='foo@example.com',
+                                      password='admin')
+        self.client.login(username='admin', password='admin')
+
+    def test_action_publish_snippet(self):
+        snippet = SnippetFactory(published=False)
+        self.assertFalse(snippet.published)
+        snippet_action_url = '/admin/base/snippet/{}/actions/publish_object/'.format(snippet.pk,)
+        self.client.get(snippet_action_url)
```

Comment:
Hmm.. didn't notice this before, but it's odd that the action can be taken using a GET request. Should really be a POST to modify an object.

Reply:
That's a very good point and I'm immediately returning this to WIP. It seems that this is how django-object-actions implements things. I'll re-think the approach